shared/lib/multichain/chain-agnostic-permission.ts

```diff
@@ -0,0 +1,228 @@
+import {
+  Caip25CaveatType,
+  Caip25CaveatValue,
+  InternalScopesObject,
+  InternalScopeString,
+  KnownSessionProperties,
+  NormalizedScopesObject,
+  parseScopeString,
+} from '@metamask/chain-agnostic-permission';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import {
+  CaipAccountId,
+  CaipChainId,
+  CaipNamespace,
+  parseCaipAccountId,
+} from '@metamask/utils';
+import { isEqualCaseInsensitive } from '../../modules/string-utils';
+
+// Helpers to be adapted and moved to @metamask/chain-agnostic-permission
```

Comment:
What's the reasoning for not doing this now? Instead of committing extra code here?

Reply:
time constraints

Reply:
Requesting leniency here with a fast follow post merge. I understand this is not best practice or ideal 🙏 

This is tracked elsewhere, but will track it in this ticket as well https://github.com/MetaMask/MetaMask-planning/issues/4599